### PR TITLE
Prepare 0.3.1 Release

### PR DESCRIPTION
**Pull request by psalz**
Here's all the [changes between v0.3.0 and current master](https://github.com/celerity/celerity-runtime/compare/v0.3.0...master).

The CMake restructuring of the examples is hardly a bugfix; I've decided to just leave it out of the changelog altogether 💁‍♂️.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,22 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+- Fix data race in `task_manager::notify_horizon_executed`. (f641bc)
+- Don't rely on static destruction order for `user_benchmarker`. (d1c9e51)
+- Restructure `wave_sim` example to avoid host side race condition for certain
+  `--sample-rate` configurations. (d226b95)
+- Hard-code paths for CMake dependencies in installed Celerity config to avoid
+  mismatches. (4e88657)
+
 ## [0.3.0] - 2021-11-16
 
 We recommend using the following SYCL versions with this release:
```
[the remark this answers is not in the document]
Comment (psalz) [reply CHANGELOG.md:30]: Done

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,11 +6,31 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+deprecation warnings.
+
+### Fixed
+
+- Remove blanket-statement error message upon early buffer deallocation, which
+  in many cases is now legal. (6851145)
+- Properly apply horizons to collective host task order-dependencies. (4488724)
+- Avoid race between horizon task generation and horizon command execution.
+  (f670868)
+- Fix data race in `task_manager::notify_horizon_executed`. (f641bc)
```
Comment (PeterTh) [inline CHANGELOG.md:21]: I don't know if it needs to be clarified that this was in debug builds.

**Review by PeterTh — APPROVED** [review]

**Review by BlackMark29A — APPROVED** [review]
Looks good to me.

File: CHANGELOG.md (hunk 2 above)
Comment (psalz) [reply CHANGELOG.md]: Done!

**Pull request merged**